go/mysql/binlog_event_rbr.go

```diff
@@ -71,21 +71,25 @@ func (ev binlogEvent) TableMap(f BinlogFormat) (*TableMap, error) {
 	result.Name = string(data[pos+1 : pos+1+l])
 	pos += 1 + l + 1
 
-	// FIXME(alainjobart) this is varlength encoded.
-	columnCount := int(data[pos])
-	pos++
+	columnCount, read, ok := readLenEncInt(data, pos)
+	if !ok {
+		return nil, vterrors.Errorf(vtrpc.Code_INTERNAL, "expected column count at position %v (data=%v)", pos, data)
+	}
+	pos = read
 
-	result.Types = data[pos : pos+columnCount]
-	pos += columnCount
+	result.Types = data[pos : pos+int(columnCount)]
```

Comment:
Why do we need the type coercion here? because `readLenEncInt` returns `uint64`.
I assume this is safe only because we don't expect columnCount to ever be large enough for this coercion to turn it into a negative number.
These coercions do smell though. Do you think it's possible to clean them up?

Reply:
Go needs to always have an `int` to address within a slice. 

> I assume this is safe only because we don't expect columnCount to ever be large enough for this coercion to turn it into a negative number.

There's no way this could cause problems unless we want to also defend against actively malicious binlogs? But then we have a much broader problem here that needs an extensive effort. 

MySQL has a hard limit of 4096 columns, so that's nowhere near any level where this is a problem. 

> These coercions do smell though. Do you think it's possible to clean them up?

No, not unless we also change the signature of functions like `readLenEncInt`. But I don't think it's a practical problem given existing MySQL limits.